include/pagmo/population.hpp

```diff
@@ -170,6 +170,7 @@ class population
         m_x.reserve(m_x.size() + 1u);
         m_f.reserve(m_f.size() + 1u);
         // The rest is noexcept.
+        update_champion(x, f);
```

Comment:
The ``update_champion()`` function could throw when it copies its arguments upon invocation, so technically it could throw. So maybe you could modify the comment to say something like: "Update champion either throws before modfying anything, or completes successfully. The rest is noexcept."